src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -737,6 +738,52 @@ PhasePartition(ts::PhaseDry{FT}) where {FT<:Real} = PhasePartition(FT(0), FT(0),
 PhasePartition(ts::PhaseEquil) = PhasePartition_equil(air_temperature(ts), air_density(ts), ts.q_tot)
 PhasePartition(ts::PhaseNonEquil) = ts.q
 
+function roots_∂e_int_∂T(T::FT, e_int::FT, ρ::FT, q_tot::FT) where {FT<:Real}
+  cvm = cv_m(PhasePartition(q_tot))
+  q_vap_sat = q_vap_saturation(T, ρ)
+  λ = liquid_fraction_equil(T)
+  L = λ*FT(LH_v0) + (1-λ)*FT(LH_s0)
+  ∂q_vap_sat_∂T = q_vap_sat*L/(FT(R_v)*T^2)
+  # T0 = FT(T_min)
+  T0 = FT(T_0)
+  return cvm + ( FT(e_int_v0) + (1-λ)*FT(e_int_i0) + (T - T0) )*∂q_vap_sat_∂T
+end
+
+"""
+    saturation_adjustment_NewtonsMethod(e_int, ρ, q_tot)
+
+Compute the temperature that is consistent with
+
+ - `e_int` internal energy
+ - `ρ` (moist-)air density
+ - `q_tot` total specific humidity
+
+using Newtons method with analytic gradients.
+
+See also [`saturation_adjustment`](@ref).
+"""
+function saturation_adjustment_NewtonsMethod(e_int::FT, ρ::FT, q_tot::FT) where {FT<:Real}
+  T_1 = max(FT(T_min), air_temperature(e_int, PhasePartition(q_tot))) # Assume all vapor
```

Comment:
Why do we need to bound this by `T_min`?

Reply:
I was matching the other saturation adjustment function, which I believe had this put in to avoid low/negative temperatures when `e_int` is very low. I plan on adding a PR that tests saturation adjustment for a range of values, and if we can remove it then I'm all for it.

Reply:
Please remove the bound in both cases. It is very risky: If it ever is active, and the air is unsaturated, `T_min` is  returned as the temperature, without warning, but it would be wrong. The bound would also degrade the initial condition for Newton iterations if active. If there are convergence issues, especially for the secant method, let's deal with it in different ways.

Reply:
Will apply

Reply:
Applying this change breaks several tests:
```
@test saturation_adjustment(internal_energy_sat(200.0, ρ, q_tot), ρ, q_tot) ≈ 200.0
@test abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(200.0, ρ, q_tot), ρ, q_tot) - 200.0) < tol_T
@test saturation_adjustment(internal_energy_sat(300.0, ρ, q_tot), ρ, q_tot) ≈ 300.0
```
They end up getting domain errors (in `saturation_vapor_pressure`)

Reply:
Hmm, turns out that saturation vapor pressure call is breaking when `T_1` is passed to check whether we're at saturation or not (in `q_v_sat = q_vap_saturation(T_1, ρ)`): Inside `saturation_adjustment`, the inputs and resulting temperature are:
```
# saturation_adjustment
(e_int, PhasePartition(q_tot)) = (-143741.164702404, PhasePartition{Float64}(0.21, 0.0, 0.0))
air_temperature(e_int, PhasePartition(q_tot)) = -473.59643691525486
# q_v_sat = q_vap_saturation(T_1, ρ) # breaks due to saturation_vapor_pressure
```
So, temperatures can become _very_ negative with large negative `e_int` (which seem valid since they're computed from `internal_energy_sat` with reasonable looking values). What assumptions are there with the `internal_energy` / `saturation_vapor_pressure` equations?

Reply:
The problem here is that for the given internal energy `e_int`, allocating all water to the vapor phase leads to a physically impossible temperature (it has to be ice). So this becomes a problem at saturation at very low temperatures, with unphysically high `q_t`. 

The physical way to deal with this is to have `T_min = eps` (or `sqrt(eps)`), or something else close to absolute zero. (This will never matter in practice, so let's make the bound as small as numerically possible.)